src/libopensc/pkcs15-coolkey.c

```diff
@@ -147,9 +147,9 @@ coolkey_find_matching_cert(sc_card_t *card, sc_cardctl_coolkey_object_t *in_obj,
 static int
 coolkey_get_attribute_ulong(sc_card_t *card, sc_cardctl_coolkey_object_t *obj, CK_ATTRIBUTE_TYPE type, CK_ULONG *value)
 {
-	const u8 *val;
-	size_t val_len;
-	u8 data_type;
+	const u8 *val = malloc(sizeof(u8));
```

Comment:
please, initialize the value here with NULL. Otherwise you are leaking the memory. The `coolkey_get_attribute()` overrides whatever you pass in as the `val` attribute.

Reply:
ack